packages/nx/src/plugins/js/project-graph/build-dependencies/explicit-project-dependencies.ts

```diff
@@ -40,19 +40,28 @@ export function buildExplicitTypeScriptDependencies(
             importExpr,
             f.file
           );
+          let targetProjectName;
           if (target) {
             if (!isRoot(source) && isRoot(target)) {
               // TODO: These edges technically should be allowed but we need to figure out how to separate config files out from root
               return;
             }
 
-            res.push({
-              sourceProjectName: source,
-              targetProjectName: target,
-              sourceProjectFile: f.file,
-              type,
-            });
+            targetProjectName = target;
+          } else {
+            // treat all unknowns as npm packages, they can be eiher
+            // - mistyped local import, which has to be fixed manually
+            // - node internals, which should still be tracked as a dependency
+            // - npm packages, which are not yet installed but should be tracked
+            targetProjectName = `npm:${importExpr}`;
```

Comment:
We are simply saying, assume this to be an external (`npm`) package. If it is, this will be resolved once we install it. If it's not, that means it's a local package and will be resolved either by fixing this import or renaming the project (in which case the graph will be fixed).

Reply:
This code is only visible to `TS/JS` workspaces so it's safe to assume an `npm` package.